client/AI/AIWrapper.cpp

```diff
@@ -389,6 +413,31 @@ namespace {
         return 1;
     }
 
+    auto IsProducibleShip(int design_id, int location_id) -> bool
```

Comment:
According to current usage `IsEnqueuableShip` will be enough.  So let's keep the API small and remove this function. 

Having two functions implies that one or both cases are valid `enqueueable and not producible` and `not enqueueable and producible`.   This makes things much complex, since knowledge about that difference should be shared (via methods documentation).

Reply:
There is a difference: buildings can be producible but not enqueuable. This is used for cases like only letting the player have one of something on the queue at a particular location. The same functions were added for ships for consistency with buildings. Can be removed if you prefer, though, I suppose...

Reply:
> The same functions were added for ships for consistency with buildings.

Let's remove it from ships. It has no sense in that context.


For buildings it has sense, but at current time it will be enough to have only one of them.  And I prefer to trade flexibility for simplicity and leave a single function as well. 

If you prefer to keep them please add a documentation string:

"Return True if a building could be queued. The building could be enqueued if it is producible and not already in the queue at this location."

"Return True if a building producible.  This implies that unlocking technology is researched and location has all requirements for it (planet type, prerequired buildings are built or enqueued). To check that building is enqueuable at this location please use IsEnqueuableShip"

I am not sure that my descriptions are correct. And this is a problem a new developer will face as well.

Reply:
done?